homeassistant/components/ollama_conversation/config_flow.py

```diff
@@ -0,0 +1,169 @@
+"""Config flow for Ollama conversation integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import sys
+from types import MappingProxyType
+from typing import Any
+
+import httpx
+import ollama
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    OptionsFlow,
+)
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.selector import (
+    NumberSelector,
+    NumberSelectorConfig,
+    NumberSelectorMode,
+    ObjectSelector,
+    TemplateSelector,
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import (
+    CONF_MAX_HISTORY,
+    CONF_MODEL,
+    CONF_MODEL_OPTIONS,
+    CONF_PROMPT,
+    DEFAULT_PROMPT,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    MAX_HISTORY_NO_LIMIT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.URL)
+        ),
+        vol.Required(CONF_MODEL): cv.string,
+        vol.Required(
+            CONF_PROMPT, description={"suggested_value": DEFAULT_PROMPT}
+        ): TemplateSelector(),
+        vol.Optional(CONF_MODEL_OPTIONS): ObjectSelector(),
+        vol.Optional(
+            CONF_MAX_HISTORY, description={"suggested_value": MAX_HISTORY_NO_LIMIT}
```

Comment:
Why do we offer this to users?

Reply:
Also not ask on initial creation, maybe even never.

Reply:
Unlike OpenAI, context window size is a factor you have to take into account locally.
Models like phi become almost useless with too much in the context, I've found.

Reply:
But how will a user know?

Reply:
Looks like they can't currently: https://github.com/ollama/ollama-python/issues/84

Reply:
Limit has be set to 20 by default